jbpm-services/jbpm-kie-services/src/main/java/org/jbpm/kie/services/impl/query/preprocessor/UserAndGroupsTasksPreprocessor.java

```diff
@@ -77,8 +78,11 @@ public void preprocess(DataSetLookup lookup) {
                         addUserAndGroupsFromIdentityProvider(orgEntities, potOwner);
                     }
 
-                    it.remove();
                 }
+
+                // we need to look for recursively for the filter
```

Comment:
this comment is quite misleading to me... looks like it missing what we need to look for?

Reply:
now we have a tree. so we need to traverse recursively for the filter.
changed it